File: api/Models/Product.cs
﻿using System;
using System.Collections.Generic;
using API.Enum;
using MongoDB.Bson.Serialization.Attributes;

namespace API.Models
{
    public class Product
    {
        [BsonId]
        public Guid? Id { get; set; }
        public string? Name { get; set; }
        public int Price { get; set; }
        public string? Description { get; set; }
        public string? Image { get; set; }
        public CategoryEnum Category { get; set; }
        public List<Review> Reviews { get; set; } = new List<Review>();
        public List<string>? sizes { get; set; }
    }
}

File: frontend/src/types/product.ts
export type Product = {
  id: String;
  name: string;
  price: number;
  description?: string;
  image?: string;
  category: Category;
};

export enum Category {
  Shirt = 0,
  pants = 1,
  shoes = 2,
  accessories = 3,
  bags = 4,
  others = 5,
}
